tests/unit/Dispatcher/GetHandlerClassCest.php

```diff
@@ -30,6 +31,66 @@ class GetHandlerClassCest
     public function dispatcherGetHandlerClass(UnitTester $I)
     {
         $I->wantToTest('Dispatcher - getHandlerClass()');
-        $I->skipTest('Need implementation');
+
```

Comment:
Can we break this test into 3 separate ones (prefix/suffix/namespace) or do some combination of those in one test using a dataProvider?

Have a look at this 

```php
use Codeception\Example;

/**
 * Blah blah description
 *
 * @dataProvider getExamples
 */
public function dispatcherGetHandlerClass(UnitTester $I, Example $example)
{
    $I->wantToTest('Dispatcher - getHandlerClass() - ' . $example[0]);
    $dispatcher = new Dispatcher();
    // test the handler name
    $dispatcher->setNamespaceName('');
    $dispatcher->setHandlerSuffix('');
    $dispatcher->setControllerName($example[1]);

    $actualHandler = $dispatcher->getHandlerClass();
    $I->assertEquals($example[2], $actualHandler);
}

private function getExamples(): array
{
    return [
        ['simple',     'hello',          'Hello'],
        ['dash',       'hello-phalcon',  'HelloPhalcon'],
        ['underscore', 'hello_phalcon',  'HelloPhalcon'],
        ['camelized',  'HelloPhalcon',   'HelloPhalcon'],
        ['with slash', 'Hello\\Phalcon', 'Hello\\Phalcon'],
    ];
}
```

Reply:
The `getExamples` can return more than those so we can use it to set the namespace and suffix as well. This way we have a test that we can extend with any new data/use case we get in the future.

Reply:
@niden done.